receiver/activedirectorydsreceiver/internal/metadata/generated_status.go

```diff
@@ -7,6 +7,6 @@ import (
 )
 
 const (
-	Type      = "activedirectorydsreceiver"
+	Type      = "activedirectoryds"
```

Comment:
The type registered in factory.go is `active_directory_ds`. Unfortunately, this change is not enough to accomodate.

Reply:
Ah, you are right. This one is a bit strange with the underscores. Do you have any solutions to this in mind, I can take a look into implementing a fix if you have an idea.

Reply:
No, for now, I lean towards taking your fix, and we will need to iterate one more time for this component and a few others.

Reply:
Another solution: align type in metadata.yaml with what we need, like `active_directory_ds` in this case. And use the directory name for Scope.Name. This solution will work even with https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21469

Reply:
No, we should keep the type as is. Otherwise, it'll be a significant breaking change. I mean, we set `active_directory_ds` in metadata.yaml to have it rendered here. But the https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21469 can be solved by setting the directory in the Scope.Name field instead of this type

Reply:
I put a wrong link in my comments. I wanted to mention a solution for https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21382